superset/assets/javascripts/dashboard/components/SliceHeader.jsx

```diff
@@ -0,0 +1,120 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import moment from 'moment';
+
+import { t } from '../../locales';
+import { getExploreUrl } from '../../explore/exploreUtils';
+import EditableTitle from '../../components/EditableTitle';
+import TooltipWrapper from '../../components/TooltipWrapper';
+
+const propTypes = {
+  slice: PropTypes.object.isRequired,
+  isExpanded: PropTypes.bool,
+  formDataExtra: PropTypes.object,
+  removeSlice: PropTypes.func.isRequired,
+  updateSliceName: PropTypes.func,
+  toggleExpandSlice: PropTypes.func,
+  forceRefresh: PropTypes.func,
+};
+
+const SliceHeader = ({ removeSlice, updateSliceName, toggleExpandSlice, forceRefresh,
```

Comment:
This seems a bit big for a function component. My assumption is that those always fully render on any parent render, as opposed to `PureComponent` which only render only if state or props have shallow-changed.  I'm not sure if this is a valid concern though.